stdlib/auth/src/main/java/org/ballerinalang/auth/ldap/nativeimpl/GetLDAPScopesOfUser.java

```diff
@@ -0,0 +1,286 @@
+
+        if (membershipProperty == null || membershipProperty.length() < 1) {
+            throw new BallerinaException("Please set membershipAttribute in configuration");
+        }
+
+        String membershipValue;
+        if (nameInSpace != null) {
+            try {
+                LdapName ldn = new LdapName(nameInSpace);
+                if (LDAPConstants.MEMBER_UID.equals(ldapAuthConfig.getMembershipAttribute())) {
+                    // membership value of posixGroup is not DN of the user
+                    List rdns = ldn.getRdns();
+                    membershipValue = ((Rdn) rdns.get(rdns.size() - 1)).getValue().toString();
+                } else {
+                    membershipValue = escapeLdapNameForFilter(ldn);
+                }
+            } catch (InvalidNameException e) {
+                LOG.error("Error while creating LDAP name from: " + nameInSpace);
+                throw new UserStoreException("Invalid naming exception for : " + nameInSpace, e);
+            }
+        } else {
+            return new String[0];
+        }
+
+        searchFilter = "(&" + searchFilter + "(" + membershipProperty + "=" + membershipValue + "))";
+        String returnedAtts[] = {roleNameProperty};
+        searchCtls.setReturningAttributes(returnedAtts);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Reading roles with the membershipProperty Property: " + membershipProperty);
+        }
+
+        List<String> list = this.getListOfNames(searchBase, searchFilter, searchCtls,
+                roleNameProperty, false);
+        String[] result = list.toArray(new String[list.size()]);
+
+        if (result != null) {
+            for (String rolename : result) {
+                LOG.debug("Found role: " + rolename);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getListOfNames(String searchBases, String searchFilter,
+                                        SearchControls searchCtls, String property, boolean appendDn)
+            throws UserStoreException {
+
+        List<String> names = new ArrayList<String>();
+        DirContext dirContext = null;
+        NamingEnumeration<SearchResult> answer = null;
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Result for searchBase: " + searchBases + " searchFilter: " + searchFilter +
+                    " property:" + property + " appendDN: " + appendDn);
+        }
+
+        try {
+            dirContext = connectionSource.getContext();
+
+            // handle multiple search bases
+            String[] searchBaseArray = searchBases.split("#");
+            for (String searchBase : searchBaseArray) {
+
+                try {
+                    answer = dirContext.search(LDAPUtils.escapeDNForSearch(searchBase), searchFilter, searchCtls);
+
+                    while (answer.hasMoreElements()) {
+                        SearchResult sr = answer.next();
+                        if (sr.getAttributes() != null) {
```

Comment:
It is better to invert the condition here to reduce the number of nested blocks.
```java
if (sr.getAttributes() == null) {
    continue;
}
...
```